x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -312,13 +314,15 @@ public Collection<Object> createComponents(
         List<Object> components = new ArrayList<>();
 
         final SSLService sslService = createSSLService(environment, resourceWatcherService);
-        setLicenseService(new ClusterStateLicenseService(settings, threadPool, clusterService, getClock(), getLicenseState()));
+        LicenseService licenseService = new ClusterStateLicenseService(settings, threadPool, clusterService, getClock(), getLicenseState());
```

Comment:
The next PR will create this with a factory that is optionally provided by an SPI definition.